code/modules/tgui/tgui.dm

```diff
@@ -121,12 +121,11 @@
 
 	// Remove titlebar and resize handles for a fancy window
 	// right from the beginning
-	var/list/config_data = get_config_data();
-	if(config_data["fancy"])
-		window_options["titlebar"] = FALSE

```

Comment:
What's the point of these now if they don't reflect desired client-side state.

Reply:
Not sure what you mean. This was just an unwrapping of what that previous codeblock did.

Reply:
You're not setting window_options now, which is not a local variable.

Reply:
Oh I see what you mean, I didn't realize window_options was a tgui level var. Going to test something on this real quick.

Reply:
After some further testing and benchmarking, the usage of window_options here wasn't actually what was killing performance it was the get_config_data call. Old usage can be preserved with no observable changes in performances. Not sure how needed window_options is considering it's only used here and only these two parameters are used though. Could save some memory by killing it.

Reply:
Also that get_config_data sucks performance that hard might be worth looking into at some point later as well.